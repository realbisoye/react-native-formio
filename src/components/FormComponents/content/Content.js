import React from 'react';
import {View, Linking} from 'react-native';
import {StyleSheet} from  'react-native';
import HTMLView from 'react-native-htmlview';
import BaseComponent from '../sharedComponents/Base';
import styles from './styles';

export default class Content extends BaseComponent {

  constructor(props) {
    super(props);
    this.onLinkPress = this.onLinkPress.bind(this);
    this.getHtmlStyles = this.getHtmlStyles.bind(this);
  }

  getHtmlStyles() {
    return {
      p: {
        ...StyleSheet.flatten(styles.p),
        color: this.props.colors.textColor,
      },
    };
  }

  onLinkPress(url) {
    Linking.openURL(url)
    .catch((e) => {
      return e;
    });
  }

  render() {
    return (
      <View style={styles.content}>
        <HTMLView
          value={this.props.component.html}
          addLineBreaks={false}
          stylesheet={this.getHtmlStyles()}
          onLinkPress={this.onLinkPress}
        />
      </View>
    );
  }
}
