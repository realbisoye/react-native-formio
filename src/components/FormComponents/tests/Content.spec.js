import React from 'react';
import renderer from 'react-test-renderer';
import colors from '../../../defaultTheme/colors';
import Content from '../content/Content.js';

 describe('Content', () => {
  describe('Content field', () => {
    const component= {
      'input': false,
      'html': '<p>Test p tag and<strong>strong tag</strong></p>\n',
      'type': 'content',
      'conditional': {
        'show': '',
        'when': null,
        'eq': ''
      },
      'key': 'testContent',
      'lockKey': true
    };
    const attachToForm = jest.fn();
     it('Renders a content component', () => {
      const element = renderer.create(
        <Content
          component={component}
          colors={colors}
          attachToForm={attachToForm}
        />
      );
      expect(element).toMatchSnapshot();
    });

     it.skip('Check the html for content component', () => {
      const element = renderer.create(
        <Content
          component={component}
          colors={colors}
          attachToForm={attachToForm}
        />
      ).find('div');
      expect(element.html()).to.equal(component.html);
    });
   });
 });
