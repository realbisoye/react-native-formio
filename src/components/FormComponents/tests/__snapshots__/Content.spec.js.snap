// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Content field Renders a content component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginBottom": 0,
      "marginHorizontal": 20,
      "marginTop": 10,
    }
  }
>
  HTMLView
</View>
`;
